src/ProcessManager/RefundPaymentProcessManager.php

```diff
@@ -43,20 +49,26 @@ public function __construct(
         OrderFullyRefundedStateResolverInterface $orderFullyRefundedStateResolver,
         RelatedPaymentIdProviderInterface $relatedPaymentIdProvider,
         RefundPaymentFactoryInterface $refundPaymentFactory,
+        OrderRepositoryInterface $orderRepository,
         EntityManagerInterface $entityManager,
         MessageBusInterface $eventBus
     ) {
         $this->orderFullyRefundedStateResolver = $orderFullyRefundedStateResolver;
         $this->relatedPaymentIdProvider = $relatedPaymentIdProvider;
         $this->refundPaymentFactory = $refundPaymentFactory;
+        $this->orderRepository = $orderRepository;
         $this->entityManager = $entityManager;
         $this->eventBus = $eventBus;
     }
 
     public function next(UnitsRefunded $unitsRefunded): void
     {
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneByNumber($unitsRefunded->orderNumber());
```

Comment:
I wonder if I should find the order here or in `RefundPaymentFactory` 🤔 Especially that we are already finding the payment method by its id in the factory. Wdyt?

Reply:
In my opinion, we should pass to the factory as final data as possible, to allow it to create the desired object _and do nothing more_ 🚀